### PR TITLE
Add TAs table to the instructor view.

### PR DESCRIPTION
**Pull request by alex-kozin**
Add TA Information table to the instructors view consisting of (as per wiki [here](https://github.com/uoft-tapp/tapp/wiki/Instructor-view)) all necessary fields. Add the ability to view and create DDAHs directly from this view (same as in the DDAH table view). The TAs are filtered by position:

image

### CODE REVIEW

File: frontend/src/libs/utils.ts [hunk 1]
```diff
@@ -187,3 +187,13 @@ export function guessActiveSession(sessions: Session[]): Session | null {
...
     return null;
 }
+
+export function getReadableAssignmentStatus(assignment: Pick<Assignment, "active_offer_status">) {
+    // TODO: confirm wording to be easier for the instructor
+    switch (assignment.active_offer_status) {
+        case "accepted":
+            return "Assigned";
+        default:
+            return capitalize(assignment.active_offer_status);
```
Comment (alex-kozin) [inline frontend/src/libs/utils.ts:197]: This is a UX question of whether we need to change the status to be more readable here

File: frontend/src/views/instructor/assignments/index.tsx [hunk 2]
```diff
@@ -1,18 +1,99 @@
...
+                        if (ddah) {
+                            setPreviewDdah(ddah);
+                            setPreviewVisible(true);
+                        }
+                    }}
+                    onCreateDDAH={(assignment_id: number) => {
+                        const assignment = assignments.find(
+                            (a) => a.id === assignment_id
+                        );
+                        if (!assignment) {
```
Comment (alex-kozin) [inline frontend/src/views/instructor/assignments/index.tsx:55]: This is not yet cleaned up - some of this logic may be abstracted away in separate components, other logic can be simplified/changed.

File: frontend/src/views/instructor/assignments/tas-table.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+/**
+ * Table that displays TA information for TAs associated with a single position.
+ *
+ * @export
+ * @param {{
+ *     position_id: number;
+ * }} {
+ *     position_id,
```
Comment (alex-kozin) [inline frontend/src/views/instructor/assignments/tas-table.tsx:29]: TODO: add param descriptions

File: frontend/src/views/instructor/assignments/tas-table.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    function ViewOrCreateDDAHCell({
+        row,
+    }: {
+        row: { original: RowData };
+    }): JSX.Element | null {
+        const original = row.original;
+        if (original.ddah != null) {
+            const ddah = original.ddah;
+            // We are a real DDAH, so we want to view
```
Comment (alex-kozin) [inline frontend/src/views/instructor/assignments/tas-table.tsx:77]: TODO: here and above - clean up the comments

File: frontend/src/views/instructor/assignments/tas-table.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        }
+        return (
+            <Button
+                variant="light"
+                size="sm"
+                title={`Create DDAH for ${original.first_name} ${original.last_name}`}
+                className="py-0"
+                onClick={() => {
+                    if (onCreateDDAH) {
+                        onCreateDDAH(original.id || -1);
```
Comment (alex-kozin) [inline frontend/src/views/instructor/assignments/tas-table.tsx:102]: The id should always be defined. TODO: check that and remove the `||`

**Comment by siefkenj** [comment]
This is looking nice! I've merged #644 which includes a basic table. Please build off of that. (The view/create buttons for the DDAHs are pretty neat :-D )

**Pull request closed without merging**